cmake/vif.cmake

```diff
@@ -4,14 +4,31 @@
 # Generates USB-C VIF policies in XML format from device tree.
 set(gen_vif_script ${ZEPHYR_BASE}/scripts/generate_usb_vif/generate_vif.py)
 set(dts_compatible usb-c-connector)
-set(vif_xml ${PROJECT_BINARY_DIR}/vif.xml)
+set(vif_xml vif.xml)
+set(vif_out ${PROJECT_BINARY_DIR}/${vif_xml})
+
 set(cmd_gen_vif ${PYTHON_EXECUTABLE} ${gen_vif_script}
         --edt-pickle ${EDT_PICKLE}
         --compatible ${dts_compatible}
-        --vif-out ${vif_xml}
-        --board ${BOARD}
+        --vif-out ${vif_out}
         )
 
+if(CONFIG_GENVIF_INPUT_VIF_XML_PATH)
+    if (EXISTS ${APPLICATION_SOURCE_DIR}/${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
+        set(vif_source_xml ${APPLICATION_SOURCE_DIR}/${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
```

Comment:
Made it a mandatory parameter, if nothing gets passed the build will break

Reply:
Seems you missed this part:
> The CMake code can then be simplified.

Code stilkl suffers from the fact that if user makes a typo, like `oof.xml` instead of the correct `foo.xml` (which is the file existing on disk), then `vif_source_xml` is unset and no file argument is passed to python script.

Hence no build failure and user may not notice a mistake was made.

Reply:
Ok, if this `vif_source_xml` parameter is included in command, it will never be empty string as we are appending it to the command only when we are sure file exists. I made this parameter mandatory while reading the command by genVIF script. If it's not included in the command. it will fail the script execution, which stops the build.